### PR TITLE
UHF-468 announcement theme

### PR DESCRIPTION
**Pull request by Arkkimaagi**
# Announcement theme
https://helsinkisolutionoffice.atlassian.net/browse/UHF-468

image
image

## What was done

* Created a TWIG template for styling announcements under `templates/content/node--announcement--default.html.twig`
* Created a new SCSS file under `src/scss/components/misc/_announcement.scss` to style announcements
* Added two new mixins to end of `src/scss/library/mixins/_mixins.scss`
  * `icon-mask-image`
  * `pseudo-icon`
* Discussed with designer, ending up with some fine tuning to original specs for backend and frontend tasks.

## How to test (existing install)

* If https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/155 has not yet been merged, follow its instructions to get a working version of the notifications on your system.
* On project root run `composer require drupal/hdbt:dev-UHF-468_announcement_theme` 
* `make drush-cr`
* In [Content > Add Content > Annoucement](https://helfi.docker.so/fi/node/add/announcement) add new notifications of all three types with varying text lengths and styles, some with optional link, some without.
* See that styles match the design (note the agreed changes in [this comment](https://helsinkisolutionoffice.atlassian.net/browse/UHF-468?focusedCommentId=46865) )
  * Remember to check mobile and desktop sizes
  * Check the scss, twig codes, their locations and generic coding style (My first bigger commit on project 🙈 )

Also check this PR: City-of-Helsinki/drupal-helfi-platform-config#155

### CODE REVIEW

File: src/scss/components/misc/_announcement.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+}
+
+.announcenemnt__content-body {
+  & *:first-child {
+    margin-top: 0;
+  }
+}
+
+.announcement__link a::after {
```
Comment (teroelonen) [inline src/scss/components/misc/_announcement.scss]: Added some spacing here after discussing with Sauli.

**Review by teroelonen — APPROVED** [review]
Some small things to consider, but nothing that would stop merging in my opinion.

File: src/scss/components/misc/_announcement.scss [hunk 2]
```diff
@@ -0,0 +1,69 @@
+.announcement {
+  --announcement-bg-color: #{$color-silver-light}; // Defaults to .announcement--notification styles
+  --announcement-bg-size: #{$spacing-double};
+  --announcement-color: #{$color-info}; // Defaults to .announcement--notification styles
+  align-items: flex-start;
+  background: linear-gradient(90deg, var(--announcement-color) 0, var(--announcement-color) var(--announcement-bg-size), var(--announcement-bg-color) var(--announcement-bg-size));
+  background-color: var(--announcement-bg-color);
+  box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.1);
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/scss/components/misc/_announcement.scss:8]: Indeed, I was wondering how we have handled the box-shadows elsewhere, I was not able to find a "global" box-shadow variable or such. Or is this the first actual usecase for such?

File: src/scss/library/mixins/_mixins.scss [hunk 3]
```diff
@@ -65,3 +65,27 @@
     }
   }
 }
+
+/**
+ * Allow changing icon easily on hover, etc.
+ */
+@mixin icon-mask-image( $icon ) {
+  mask-image: url("../icons/hds/ui/" + $icon + ".svg");
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/scss/library/mixins/_mixins.scss:73]: Given that information I'd add the `ui/` and `some/` part to the string to deliver to the mixin. If the separation between some and ui is a wanted separation, we should follow it in code. I will commit a new version where this is fixed.

File: src/scss/components/misc/_announcement.scss (hunk 1 above)
Comment (Arkkimaagi) [reply src/scss/components/misc/_announcement.scss]: ok

File: src/scss/components/misc/_announcement.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/scss/components/misc/_announcement.scss]: Fixed the `0px`

File: templates/content/node--announcement--default.html.twig [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  {% set block_modifier = 'announcement--attention' %}
+{% else %}
+  {% set block_modifier = 'announcement--notification' %}
+{% endif %}
+
+{% set link = content.field_announcement_link[0] %}
+
+<div class="announcement {{ block_modifier }}" data-nodeid="{{ node.id }}">
+  <div class="announcement__content">
+    <span class="announcenemnt__content-body">{{ content.body }}</span>
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply templates/content/node--announcement--default.html.twig]: Given that the admin ui allows empty messages (weird IMO), and as we should soon have link with description supported, this is an option. I added condition to handle this case.

File: src/scss/library/mixins/_mixins.scss [hunk 5]
```diff
@@ -65,3 +65,27 @@
...
+}
+
+/**
+ * Add pseudo element (::before, ::after) styling for icons
+ */
+@mixin pseudo-icon($icon, $size: $spacing, $color: $color-black) {
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/scss/library/mixins/_mixins.scss:79]: Indeed, I did not want to overload this PR with the conversion, especially as we're nearing the release soon.

**Pull request merged**